vllm/multimodal/base.py

```diff
@@ -43,6 +43,16 @@ def try_concat(
         *,
         device: torch.types.Device,
     ) -> BatchedTensors:
+        # Avoid initializing CUDA too early
+        import torch
+
+        # may be list rather than tensors
+        if not isinstance(tensors[0], torch.Tensor):
+            new_tensors = []
+            for new_tensor in tensors:
+                for new_t in new_tensor:
+                    new_tensors.append(new_t.to(device))
+            return new_tensors
```

Comment:
Can you override the image processor for your model to convert the list into `torch.Tensor`? We should avoid modifying the existing code if possible.

Reply:
Sry, But the images might not in the same shape, so it cloud not be concat...

Reply:
Why the tensors need to concat here? For multi-image inputs? And each image may split to multi-patches. Things here are concatenating these patches, am I right?

Reply:
The concatenation is performed along the batch dimension. If for each example in the batch, the tensor has different size, then a list is returned, otherwise a stacked tensor is returned. This is already handled by the existing logic.

Reply:
> The concatenation is performed along the batch dimension. If for each example in the batch, the tensor has different size, then a list is returned, otherwise a stacked tensor is returned. This is already handled by the existing logic.

The shapes of image processor's outputs are indeed inconsistent...So the `tensors[0]`, which you want to extract its shape, might be a tensor...